sdccc/src/main/java/com/draeger/medical/sdccc/tests/glue/direct/DirectSubscriptionHandlingTest.java

```diff
@@ -293,19 +335,40 @@ void testRequirementR0036() throws InterruptedException {
                                     ActionConstants.ACTION_EPISODIC_OPERATIONAL_STATE_REPORT,
                                     WsdlConstants.SERVICE_STATE_EVENT,
                                     MessageGeneratingUtil::getStateEventService,
-                                    report);
+                                    report,
+                                    false);
                         },
                         () -> {
                             // does not need to be triggered in this test.
-                        }));
+                        }) {
+                    @Override
+                    public boolean doesNotificationBodyBelongToThisReport(final Object notificationBody) {
+                        return notificationBody instanceof EpisodicOperationalStateReport;
+                    }
+                });
+        final List<ReportTestData> reports = new ArrayList<>();
+        reports.add(triggerableReport);
+        reports.addAll(otherReports);
 
         subscribeToAllReports(reports);
 
         checkSubscriptionsAreActive(reports);
 
         triggerReportAndIntentionallyFailReceivingIt(triggerableReport);
 
-        checkThatAllSubscriptionsAreCancelled(reports);
+        checkThatSubscriptionsHaveBeenCancelled(List.of(triggerableReport));
+        checkThatOtherSubscriptionsHaveNotBeenCancelled(otherReports);
+    }
+
+    private void checkThatOtherSubscriptionsHaveNotBeenCancelled(final List<ReportTestData> reports) {
+        for (ReportTestData report : reports) {
+            if (report.getSubscription() != null) {
```

Comment:
would be better to treat null as an error here instead of ignoring it

Reply:
you can still verify that at least one is not equal to null

Reply:
same for checkSubscriptionsAreActive

Reply:
added a check that lets the Test fail when EpisodicContextReport is not supported.